aws/resource_aws_worklink_fleet.go

```diff
@@ -0,0 +1,399 @@
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringLenBetween(1, 204800),
+						},
+					},
+				},
+			},
+			"company_code": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_updated_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"optimize_for_end_user_location": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+		},
+	}
+}
+
+func resourceAwsWorkLinkFleetCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).worklinkconn
+
+	input := &worklink.CreateFleetInput{
+		FleetName:                  aws.String(d.Get("name").(string)),
+		OptimizeForEndUserLocation: aws.Bool(d.Get("optimize_for_end_user_location").(bool)),
+	}
+
+	if v, ok := d.GetOk("display_name"); ok {
+		input.DisplayName = aws.String(v.(string))
+	}
+
+	resp, err := conn.CreateFleet(input)
+	if err != nil {
+		return fmt.Errorf("Error creating Worklink Fleet: %s", err)
+	}
+
+	d.SetId(aws.StringValue(resp.FleetArn))
+	return resourceAwsWorkLinkFleetUpdate(d, meta)
+}
+
+func resourceAwsWorkLinkFleetRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).worklinkconn
+
+	resp, err := conn.DescribeFleetMetadata(&worklink.DescribeFleetMetadataInput{
+		FleetArn: aws.String(d.Id()),
+	})
+	if err != nil {
+		if isAWSErr(err, worklink.ErrCodeResourceNotFoundException, "") {
+			log.Printf("[WARN] Worklink Fleet (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error describe worklink fleet: %s", err)
+	}
+
+	d.Set("arn", d.Id())
+	d.Set("name", aws.StringValue(resp.FleetName))
+	d.Set("display_name", aws.StringValue(resp.DisplayName))
+	d.Set("optimize_for_end_user_location", aws.BoolValue(resp.OptimizeForEndUserLocation))
+	d.Set("company_code", aws.StringValue(resp.CompanyCode))
+	d.Set("created_time", resp.CreatedTime.Format(time.RFC3339))
+	if resp.LastUpdatedTime != nil {
+		d.Set("last_updated_time", resp.LastUpdatedTime.Format(time.RFC3339))
+	}
+	auditStreamConfigurationResp, err := conn.DescribeAuditStreamConfiguration(&worklink.DescribeAuditStreamConfigurationInput{
+		FleetArn: aws.String(d.Id()),
+	})
+	if err != nil {
+		if !isAWSErr(err, worklink.ErrCodeResourceNotFoundException, "") {
+			return fmt.Errorf("Error describe worklink audit stream configuration: %s", err)
+		}
+	}
+	d.Set("audit_stream_arn", auditStreamConfigurationResp.AuditStreamArn)
+
+	companyNetworkConfigurationResp, err := conn.DescribeCompanyNetworkConfiguration(&worklink.DescribeCompanyNetworkConfigurationInput{
+		FleetArn: aws.String(d.Id()),
+	})
+	if err != nil {
+		return fmt.Errorf("Error describe worklink company network configuration: %s", err)
+	}
+	d.Set("network", flattenWorkLinkNetworkConfigResponse(companyNetworkConfigurationResp))
```

Comment:
When using `d.Set()` with aggregate types (TypeList, TypeSet, TypeMap), [we should perform error checking to prevent issues where the code is not properly able to set the Terraform state](https://www.terraform.io/docs/extend/best-practices/detecting-drift.html#error-checking-aggregate-types). e.g.

```suggestion
	if err := d.Set("network", flattenWorkLinkNetworkConfigResponse(companyNetworkConfigurationResp)); err != nil {
		return fmt.Errorf("error setting network: %s", err)
	}
```

Reply:
Fixed.